backend/siarnaq/api/compete/views.py

```diff
@@ -387,6 +388,80 @@ def scrimmage(self, request, pk=None, *, episode_id):
         serializer = self.get_serializer(page, many=True)
         return self.get_paginated_response(serializer.data)
 
+    def get_historical_rating_ranking(self, episode_id, queryset, limit=None):
+        """
+        Retrieve historical ratings for teams in a specific episode, sorted by
+        highest rating.
+
+        This function processes a set of matches to calculate and return historical
+        ratings for teams participating in a given episode. The results are ordered
+        by the maximum rating achieved, in descending order.
+
+        Parameters:
+        - episode_id (int): The identifier of the episode to filter teams by.
+        - queryset (QuerySet): A collection of match objects used to compute
+                               historical ratings.
+        - limit (int, optional): The maximum number of team ratings to return.
+
+        Returns:
+        - List[Dict]: A list of dictionaries containing team information and
+                      their historical ratings, sorted by highest rating in
+                      descending order. The number of items is capped by
+                      the 'limit' parameter if provided.
+        """
+        has_invisible = self.get_queryset().filter(
+            participants__team__status=TeamStatus.INVISIBLE
+        )
+        matches = (
+            (
+                queryset.filter(episode=episode_id)
+                .filter(tournament_round__isnull=True)
+                .exclude(pk__in=Subquery(has_invisible.values("pk")))
+                .filter(is_ranked=True)
+                .filter(participants__rating__isnull=False)
+            )
+            .all()
+            .order_by("-created")
+        )
+
+        # query aggregate rating history per each team order by
+        # max_rating and limit by 10
+        matching_participants = MatchParticipant.objects.all().filter(match__in=matches)
+        rating_history = (
+            matching_participants.values("team_id")
+            .annotate(
+                timestamps_list=ArrayAgg(
+                    F("match__created"), ordering="match__created"
+                ),
+                ratings_pk_list=ArrayAgg(F("rating__pk"), ordering="match__created"),
+                max_rating=Max("rating__value"),
+            )
+            .all()
+            .order_by("-max_rating")[:limit]
```

Comment:
is there no `.limit()` or similar django method?

Reply:
doesn't seem like it